### PR TITLE
Fix: Prevent flag injection when query starts with dash (ENG-2092)

### PR DESCRIPTION
**Pull request by K-Mistele**
## What problem(s) was I solving?

Fixed a critical bug where Claude CLI sessions would fail when user queries started with a dash (`-`) character. The CLI's argument parser was misinterpreting these queries as command-line flags, causing "unknown option" errors that crashed Claude sessions.

Example error from production:
```
claude process failed: claude error: error: unknown option '-one thing, but we'll just include some notes in the report...'
```

Related issues:
- [ENG-2092](https://linear.app/humanlayer/issue/ENG-2092/claude-launched-with-unsafe-stringing-if-prompt-starts-with-it-treated): claude launched with unsafe stringing, if prompt starts with `-` it treated like a flag

### Claude CLI Flag Injection Fix
- **Before**: Queries starting with `-` or `--` would cause Claude sessions to crash with "unknown option" errors
- **After**: All queries work correctly, regardless of whether they start with dashes or look like flags
- Examples that now work:
  - `-one thing - just do one thing`
  - `--help` (processes as text instead of showing help)
  - `--model opus` (processes as query instead of trying to set model)

### Claude CLI Fix (`claudecode-go/client.go`)
The core issue was argument ordering. The `--print` flag and query were being added at the beginning of the argument list, before other flags like `--model`, `--resume`, etc.

**Solution**: Use the POSIX standard `--` separator to tell the CLI parser to stop interpreting flags:
1. Moved query argument construction to the END of `buildArgs()` function
2. Added `--` separator between `--print` and the query value
3. Ensured command structure: `claude <all-flags> --print -- <query>`

**Why this works**: The `--` separator is a universal CLI convention that signals "everything after this point is data, not flags." This is supported by all major argument parsing libraries.

### Test Coverage (`claudecode-go/client_internal_test.go`)
Added comprehensive test suite with 2 test functions covering:
- Basic dash-prefixed query handling (6 test cases)
- Complex configurations with multiple flags (3 test cases)
- Verification of correct argument ordering
- Edge cases: single dash, double dash, dash in middle, flag-like queries

Tests use internal package access to test the private `buildArgs()` method directly, ensuring correctness before command execution.

## How to verify it

- [x] I have ensured `make check test` passes (all 50 Go tests, 341 daemon tests, UI tests pass)

### Manual Testing

1. Test dash-prefixed queries via CodeLayer UI:
   ```
   Send message: "-one thing - just do one thing"
   Expected: Claude processes the query successfully without errors
   ```

2. Test via command line:
   ```bash
   # Should work with various flags
   claude --model sonnet --print -- "-help me"
   claude --output-format json --print -- "--test"

   # Verify old syntax fails (without --)
   claude --print "-one thing"  # Should fail with "unknown option"
   ```

3. Verify backwards compatibility:
   - Normal queries without dashes continue to work
   - All existing flags (`--model`, `--resume`, `--output-format`, etc.) work correctly
   - Query is always processed last, after all flags

## Description for the changelog

Fixed critical bug (ENG-2092) where Claude CLI sessions would crash when user queries started with dash characters. Added POSIX `--` separator to properly handle queries like "-one thing" or "--help" as text instead of flags. Includes comprehensive test coverage.

### CODE REVIEW

File: claudecode-go/client.go [hunk 1]
```diff
@@ -303,6 +298,15 @@ func (c *Client) buildArgs(config SessionConfig) ([]string, error) {
 		args = append(args, "--verbose")
 	}
 
+	// Always use print mode for SDK - MUST be the last flag before --
+	// The -- separator tells the CLI parser to stop interpreting flags
+	// Correct order: <all flags> --print -- <query>
+	if config.Query != "" {
```
Comment (dexhorthy) [inline claudecode-go/client.go:304]: i mean technical if `config.Query == ""` then we have nothing to do and claude will fail but...the old code did that so fine with this
